### PR TITLE
Add commitizen to make it easier to follow the Commit Message Guidelines

### PR DESCRIPTION
**Pull request by rikhall1515**
## Description
This change makes it easier to follow the conventional commits standard by 
automatically using the commitizen CLI whenever you execute `git commit` (no -m flag).

## Validation
I ran `git commit`, used the CLI, and commitizen worked as it should.

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing.
- [x] I have run `npx turbo build` to check if the website builds without errors.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -111,10 +111,12 @@ for getting things done and landing your contribution.
 
 10. Once you're happy with your changes, add and commit them to your branch, then push the branch to your fork.
 
+    Note that this repository is now commitizen friendly, which means you should just use `git commit` with no flags.
```
Comment (araujogui) [inline CONTRIBUTING.md]:
```suggestion
    Note that this repository is commitizen friendly, which means you should just use `git commit` with no flags.
```

File: commitlint.config.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
+import type { UserConfig } from '@commitlint/types';
+
+const Configuration: UserConfig = {
```
Comment (araujogui) [inline commitlint.config.ts]: Maybe we should delete these commented config options

**Review by araujogui — CHANGES_REQUESTED** [review]

**Comment by rikhall1515** [comment]
I'm honestly debating if I should even keep commitlint in this change.

I've changed the config so that it's not terribly strict about the rules. It will allow
sentence case, lower case, and anything in-between (in case of accidental typos).

More strictness just leads to a terrible developer experience because there is no
easy way of integrating linting into the commit message file (that works on all IDEs).
If you close the file, have it be linted, and then it fails; now you must redo the whole 
thing. I've checked and there isn't an easy way around that.

Also, you get 80% of the result anyway if you just use `git commit` and let commitizen
do most of the work. Commitlint at this point only exists to prevent cases where the user
intentionally avoids commitizen.

**Comment by ovflowd** [comment]
> lso, you get 80% of the result anyway if you just use `git commit` and let commitizen
> do most of the work. Commitlint at this point only exi

I feel that if we are so permissive for commit messages, we lose the reasoning behind CommitLint, no?

I do believe adding Semantic Versioning would be great tho!!

**Comment by rikhall1515** [comment]
> > lso, you get 80% of the result anyway if you just use `git commit` and let commitizen
> > do most of the work. Commitlint at this point only exi
> 
> I feel that if we are so permissive for commit messages, we lose the reasoning behind CommitLint, no?
> 
> I do believe adding Semantic Versioning would be great tho!!

Yeah you're right, I'll update the PR so it's just commitizen in that case.

File: commitlint.config.ts (hunk 2 above)
Comment (Jean681729) [reply commitlint.config.ts]: Resolved

**Review by araujogui — APPROVED** [review]
LGTM. Thanks for you first contribution!

**Review by aymen94 — APPROVED** [review]

**Comment by aymen94** [comment]
LGTM, @rikhall1515 thanks for contribution.

File: package.json [hunk 3]
```diff
@@ -34,7 +34,8 @@
     "test:unit": "cross-env NODE_NO_WARNINGS=1 jest",
     "test:unit:watch": "npm run test:unit -- --watch",
     "test": "turbo test:unit",
-    "prepare": "husky"
+    "prepare": "husky",
+    "cm": "cz"
```
Comment (bmuenzenmeyer) [inline package.json]: is this just for convenience if someone wants to call it directly? it seems like the primary use case is via husky git hook

Comment (rikhall1515) [reply package.json]: You're right, the primary use case is with the husky git hook. 

If you want I can remove the script?

Comment (ovflowd) [reply package.json]: Go ahead :)

Comment (rikhall1515) [reply package.json]: Okay! All done.

**Comment by rikhall1515** [comment]
I realize my most recent commit was perhaps unnecessary after reading the CONTRIBUTING.md a bit closer when it comes to rebasing/merging recent changes.

I assume there's nothing else I need to do right now in order for this PR to be merged?

**Comment by bmuenzenmeyer** [comment]
> I assume there's nothing else I need to do right now in order for this PR to be merged?

I want to test it locally - also, there is a merge conflict that you cool look into

**Comment by rikhall1515** [comment]
Could someone check what the details are for the deployment? What I did is I copied the current upstream/main package-lock.json and package.json - then I did `npm ci` followed by adding the commitizen and cz-conventional-commits dependencies.

It should just work because I have not modified any files relating to the actual source code.

Definitely need to squash the changes on this PR 😅

**Comment by ovflowd** [comment]
Vercel is not happy with the changes :thinking: 

Can we get a rebase here? That might (?) help

**Comment by rikhall1515** [comment]
> Vercel is not happy with the changes 🤔
> 
> Can we get a rebase here? That might (?) help

@ovflowd I've re-opened the same PR so now it only has one commit.

**Pull request closed without merging**